contrib/NeonVM/neonvm.c

```diff
@@ -0,0 +1,820 @@
+    error("not implemented");
+}
+
+void exec_EQN()
+{
+    error("not implemented");
+}
+
+void exec_NEN()
+{
+    error("not implemented");
+}
+
+void exec_LTN()
+{
+    error("not implemented");
+}
+
+void exec_GTN()
+{
+    error("not implemented");
+}
+
+void exec_LEN()
+{
+    error("not implemented");
+}
+
+void exec_GEN()
+{
+    error("not implemented");
+}
+
+void exec_EQS()
+{
+    error("not implemented");
+}
+
+void exec_NES()
+{
+    error("not implemented");
+}
+
+void exec_LTS()
+{
+    error("not implemented");
+}
+
+void exec_GTS()
+{
+    error("not implemented");
+}
+
+void exec_LES()
+{
+    error("not implemented");
+}
+
+void exec_GES()
+{
+    error("not implemented");
+}
+
+void exec_EQA()
+{
+    error("not implemented");
+}
+
+void exec_NEA()
+{
+    error("not implemented");
+}
+
+void exec_EQD()
+{
+    error("not implemented");
+}
+
+void exec_NED()
+{
+    error("not implemented");
+}
+
+void exec_EQP()
+{
+    error("not implemented");
+}
+
+void exec_NEP()
+{
+    error("not implemented");
+}
+
+void exec_ANDB()
+{
+    error("not implemented");
+}
+
+void exec_ORB()
+{
+    error("not implemented");
+}
+
+void exec_NOTB()
+{
+    error("not implemented");
+}
+
+void exec_INDEXAR()
+{
+    error("not implemented");
+}
+
+void exec_INDEXAW()
+{
+    error("not implemented");
+}
+
+void exec_INDEXAV()
+{
+    error("not implemented");
+}
+
+void exec_INDEXAN()
+{
+    error("not implemented");
+}
+
+void exec_INDEXDR()
+{
+    error("not implemented");
+}
+
+void exec_INDEXDW()
+{
+    error("not implemented");
+}
+
+void exec_INDEXDV()
+{
+    error("not implemented");
+}
+
+void exec_INA()
+{
+    error("not implemented");
+}
+
+void exec_IND()
+{
+    error("not implemented");
+}
+
+void exec_CALLP(struct tagTExecutor *self)
+{
+    uint32_t val = (self->object->code[self->ip+1] << 24) | (self->object->code[self->ip+2] << 16) | (self->object->code[self->ip+3] << 8) | self->object->code[self->ip+4];
+    self->ip += 5;
+    const unsigned char *func = self->object->strtable[val];
+
+    if (strcmp((const char*)func, "print") == 0) {
+        const char *s = pop(&self->stack);
+        printf("%s\n", s);
+    } else {
+        error("exec_CALLP(): Invalid function call");
+    }
+}
+
+void exec_CALLF()
+{
+    error("not implemented");
+}
+
+void exec_CALLMF()
+{
+    error("not implemented");
+}
+
+void exec_CALLI()
+{
+    error("not implemented");
+}
+
+void exec_JUMP()
+{
+    error("not implemented");
+}
+
+void exec_JF()
+{
+    error("not implemented");
+}
+
+void exec_JT()
+{
+    error("not implemented");
+}
+
+void exec_JFCHAIN()
+{
+    error("not implemented");
+}
+
+void exec_DUP()
+{
+    error("not implemented");
+}
+
+void exec_DUPX1()
+{
+    error("not implemented");
+}
+
+void exec_DROP()
+{
+    error("not implemented");
+}
+
+void exec_RET(struct tagTExecutor *self)
+{
+    const char *code = pop(&self->callstack);
+    self->ip = (code[3] << 24) | (code[2] << 16) | (code[1] << 8) | code[0];

```

Comment:
Rather than storing pointers to integers on the callstack, it would be better to store the integers themselves. This will definitely be a problem in the future if left unchanged.